src/shm_sub.c

```diff
@@ -2270,26 +2271,52 @@ sr_shmsub_rpc_internal_call_callback(sr_conn_ctx_t *conn, const struct lyd_node
         return err_info;
     }
 
-    /* collect all required modules */
     LY_LIST_FOR(lyd_child(input), child) {
-        /* get LY module */
-        ly_mod = ly_ctx_get_module_implemented(conn->ly_ctx, lyd_get_value(child));
-        if (!ly_mod) {
-            sr_errinfo_new(&err_info, SR_ERR_NOT_FOUND, "Module \"%s\" was not found in sysrepo.", lyd_get_value(child));
-            goto cleanup;
-        } else if (!strcmp(ly_mod->name, "sysrepo")) {
-            sr_errinfo_new(&err_info, SR_ERR_INVAL_ARG, "Internal module \"%s\" cannot be reset to factory-default.",
-                    lyd_get_value(child));
-            goto cleanup;
-        }
+        if (!strcmp(LYD_NAME(child), "modules")) {
```

Comment:
Just a minor improvement, you can avoid one loop using `lyd_find_path()` (which is not using a set) to find first `modules` and then `datastores` nodes, then you can iterate over all their children.

Reply:
updated.

I've added a TODO where I'm not sure why "child->next" is required instead of "child" ... do you know what I'm doing wrong?

Reply:
Why is `child->next` needed? I do not think it is.

Reply:
there are two modules nodes found using `sr_lyd_find_path` and only the one used with `->next` has the children